keylime/src/tpm.rs

```diff
@@ -225,6 +305,345 @@ impl Context {
+                    None,
+                    Some(pcr_selection_list),
+                )
+            })
+            .unwrap();
+
+        Ok(IDevIDResult {
+            public: primary_key.out_public,
+            handle: primary_key.key_handle,
+        })
+    }
+
+    /// Mount the template for IDevID
+
+    pub(crate) fn create_idevid_public_from_default_template(
+        asym_alg: AsymmetricAlgorithm,
+        name_alg: HashingAlgorithm,
+    ) -> Result<IDevIDPublic> {
+        let obj_attrs_builder = ObjectAttributesBuilder::new()
+            .with_fixed_tpm(true)
+            .with_st_clear(false)
+            .with_fixed_parent(true)
+            .with_sensitive_data_origin(true)
+            .with_user_with_auth(true)
+            .with_admin_with_policy(true)
+            .with_no_da(false)
+            .with_encrypted_duplication(false)
+            .with_sign_encrypt(true)
+            .with_decrypt(false)
+            // restricted=0 for DevIDs
+            .with_restricted(false);
+
+        let obj_attrs = obj_attrs_builder.build()?;
+
+        let (auth_policy, key_bits, curve_id) = match name_alg {
+            HashingAlgorithm::Sha256 => (
+                IDEVID_AUTH_POLICY_SHA256[0..32].to_vec(),
+                RsaKeyBits::Rsa2048,
+                EccCurve::NistP256,
+            ),
+            HashingAlgorithm::Sm3_256 => (
+                IDEVID_AUTH_POLICY_SHA256[0..32].to_vec(),
+                RsaKeyBits::Rsa2048,
+                EccCurve::Sm2P256,
+            ),
+            HashingAlgorithm::Sha384 => (
+                IDEVID_AUTH_POLICY_SHA384[0..48].to_vec(),
+                RsaKeyBits::Rsa3072,
+                EccCurve::NistP384,
+            ),
+            HashingAlgorithm::Sha512 => (
+                IDEVID_AUTH_POLICY_SHA512[0..64].to_vec(),
+                RsaKeyBits::Rsa4096,
+                EccCurve::NistP521,
+            ),
+            _ => (
+                IDEVID_AUTH_POLICY_SHA256[0..32].to_vec(),
+                RsaKeyBits::Rsa2048,
+                EccCurve::NistP256,
+            ),
+        };
+
+        let key_builder = match asym_alg {
+            AsymmetricAlgorithm::Rsa => PublicBuilder::new()
+                .with_public_algorithm(PublicAlgorithm::Rsa)
+                .with_name_hashing_algorithm(name_alg)
+                .with_object_attributes(obj_attrs)
+                .with_auth_policy(Digest::try_from(auth_policy)?)
+                .with_rsa_parameters(
+                    PublicRsaParametersBuilder::new()
+                        .with_symmetric(SymmetricDefinitionObject::Null)
+                        .with_scheme(RsaScheme::Null)
+                        .with_key_bits(key_bits)
+                        .with_exponent(RsaExponent::default())
+                        .with_is_signing_key(obj_attrs.sign_encrypt())
+                        .with_is_decryption_key(obj_attrs.decrypt())
+                        .with_restricted(obj_attrs.restricted())
```

Comment:
I don't know what effect these last 3 have but I don't see them in the spec 7.3.4.2

Reply:
These come from 7.3.4.1, they are flags for the builder to specify if the key will be used to sign, decrypt etc